### ISSUE TITLE
QASM get_phase and get_frequency result in errors

### ISSUE THREAD

**Issue by ayush-pancholy** [oqc-community/qat#16]
**Describe the bug**
When submitting QASM strings that contain `get_phase` or `get_frequency`, I get a `TypeError: 'PulseChannelView' object is not subscriptable`. 

**Describe what you expected to happen**
I expect to be able to use these QASM functions without error - it's nice to be able to query the device data without having to be sent it externally.

**Add supporting information**
As an example QASM snippet that fails, something as simple as querying the frequency of `q0_drive` and leaving it unchanged results in an error: `set_frequency(q0_drive, get_frequency(q0_drive));`

Full stack trace:
```Traceback (most recent call last):
  File "/home/ayushp/qat_playground/qat_qasm_example.py", line 87, in <module>
    builder2 = parser.parse(
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 2114, in parse
    self.visit(parsed)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 419, in visit
    return self._visit_tree(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 427, in _visit_tree
    return f(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 437, in __default__
    return self.visit_children(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in visit_children
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in <listcomp>
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 427, in _visit_tree
    return f(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1711, in quantum_gate_call
    self._call_gate(name, method_args)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1688, in _call_gate
    cali_results = self.visit_children(body)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in visit_children
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in <listcomp>
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 427, in _visit_tree
    return f(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 437, in __default__
    return self.visit_children(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in visit_children
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 430, in <listcomp>
    return [self._visit_tree(child) if isinstance(child, Tree) else child
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 427, in _visit_tree
    return f(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1906, in extern_or_subroutine_call
    args = self.transform_to_value(tree.children[1])
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1159, in transform_to_value
    return [
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1160, in <listcomp>
    self.transform_to_value(val, walk_variable, return_variable)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1146, in transform_to_value
    return self.visit(node)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 419, in visit
    return self._visit_tree(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/lark/visitors.py", line 427, in _visit_tree
    return f(tree)
  File "/home/ayushp/miniconda3/envs/qat/lib/python3.9/site-packages/qat/purr/integrations/qasm.py", line 1946, in extern_or_subroutine_call
    channel, _ = self._validate_freq_args(args[0])
TypeError: 'PulseChannelView' object is not subscriptable```

**Comment by ayush-pancholy** [comment]
Update: This bug can be fixed by merging #17.

**Comment by ayush-pancholy** [comment]
Bug should be fixed with #18 merged!
